clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -474,6 +474,9 @@ private long trySend(long now) {
                 if (client.ready(node, now)) {
                     client.send(request, now);
                     iterator.remove();
+                } else {
```

Comment:
I'm wondering if it is better to do the `client.ready()` check outside the `while` loop? If the client is not ready, it will not become ready during the while loop, and vice versa. WDYT @hachikuji ?

Reply:
Hi @guozhangwang, thank you for your review. IMHO, `client.ready()` also checks the number of in-flight requests. and the client will become not ready when the number of in-flight requests reaches `maxInFlightRequestsPerConnection`.

Reply:
Good point. Makes sense.